src/spring-cloud/azext_spring_cloud/_app_factory.py

```diff
@@ -23,10 +24,37 @@ def _format_properties(self, **kwargs):
         kwargs['temporary_disk'] = self._load_temp_disk(**kwargs)
         return models.AppResourceProperties(**kwargs)
 
-    def _format_identity(self, assign_identity=None, **_):
-        if assign_identity is not None:
-            assign_type = 'systemassigned' if assign_identity else 'None'
-            return models.ManagedIdentityProperties(type=assign_type)
+    def _format_identity(self, system_assigned=None, user_assigned=None, **_):
+        target_identity_type = self._get_identity_assign_type(system_assigned, user_assigned)
+        user_identity_payload = self._get_user_identity_payload(target_identity_type, user_assigned)
+        identity_props = None
+        if target_identity_type != models_20220301preview.ManagedIdentityType.NONE:
+            identity_props = models_20220301preview.ManagedIdentityProperties()
+            identity_props.type = target_identity_type
+            identity_props.user_assigned_identities = user_identity_payload
+        return identity_props
+
+    def _get_identity_assign_type(self, system_assigned=None, user_assigned=None):
+        target_identity_type = models_20220301preview.ManagedIdentityType.NONE
+        if system_assigned and user_assigned:
+            target_identity_type = models_20220301preview.ManagedIdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED
+        elif system_assigned:
+            target_identity_type = models_20220301preview.ManagedIdentityType.SYSTEM_ASSIGNED
+        elif user_assigned:
+            target_identity_type = models_20220301preview.ManagedIdentityType.USER_ASSIGNED
+        return target_identity_type
+
+    def _get_user_identity_payload(self, target_identity_type, user_assigned=None):
+        if target_identity_type in (models_20220301preview.ManagedIdentityType.NONE,
+                                    models_20220301preview.ManagedIdentityType.SYSTEM_ASSIGNED):
+            return None
+        user_identity_payload = {}
+        if user_assigned:
+            for user_identity_resource_id in user_assigned:
+                user_identity_payload[user_identity_resource_id] = models_20220301preview.UserAssignedManagedIdentity()
```

Comment:
nit: do we need to check the format here? To reduce potential data inconsistent when sending to backend?